### PR TITLE
Upgrade to React 0.14 and corresponding Fluxible packages

### PR DESCRIPTION
**Pull request by mridgway**
Also removed some of the unused functionality.

### CODE REVIEW

**Comment by mridgway** [comment]
Updated to full release from RC. Ready for review.

**Comment by yahoocla** [comment]
CLA is valid!

File: app/templates/components/Application.js [hunk 1]
```diff
@@ -27,15 +28,13 @@ class Application extends React.Component {
     }
 }
 
-export default handleHistory(provideContext(connectToStores(
+export default provideContext(handleHistory(connectToStores(
```
Comment (Vijar) [inline app/templates/components/Application.js]: Do you need provideContext if you're using createElementWithContext?

**Comment by Vijar** [comment]
Other than my comment, :ship:

File: app/templates/components/Application.js (hunk 1 above)
Comment (mridgway) [inline app/templates/components/Application.js]: You don't need it, but it's still the recommended way to do it. Without it, `createElementWithContext` will use a `FluxibleComponent` but that does not support custom context types.

**Pull request merged**